apps/dashboard/src/hooks/use-invocation-queue.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useRef } from 'react';
+
+type CallbackFunction = () => Promise<unknown>;
+
+export function useInvocationQueue<T extends CallbackFunction = CallbackFunction>({
```

Comment:
The hook that queues all the autosave api calls and runs them one by one in order, to guarantee that there wouldn't be any race conditions between users changes.